platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponent.java

```diff
@@ -422,6 +459,55 @@ public void activateLocationComponent(@NonNull Context context, @NonNull Style s
     applyStyle(options);
   }
 
+  public void activateLocationComponent(@NonNull LocationComponentActivationOptions
+                                          locationComponentActivationOptions) {
+
+    Log.d(TAG, "locationComponentActivationOptions context packageName = " + locationComponentActivationOptions.context().getPackageName());
+    Log.d(TAG, "locationComponentActivationOptions style URL = " + locationComponentActivationOptions.style().getUrl());
+    Log.d(TAG, "locationComponentActivationOptions styleRes = " + locationComponentActivationOptions.styleRes());
+
+    if (locationComponentActivationOptions.context() != null
+      && locationComponentActivationOptions.style() != null
+      && locationComponentActivationOptions.locationComponentOptions() != null) {
+      initialize(locationComponentActivationOptions.context(), locationComponentActivationOptions.style(),
+        locationComponentActivationOptions.locationComponentOptions());
+    }
+
+    if (locationComponentActivationOptions.locationEngineRequest() != null) {
+      setLocationEngineRequest(locationComponentActivationOptions.locationEngineRequest());
+    }
+
+    if (locationComponentActivationOptions.locationEngine() != null) {
+      setLocationEngine(locationEngine);
+    } else if (locationComponentActivationOptions.useDefaultLocationEngine()) {
+      initializeLocationEngine(locationComponentActivationOptions.context());
+    } else {
+      setLocationEngine(null);
+    }
+
+    // TODO: How to do some sort of null check on an int for locationComponentActivationOptions.styleRes() ?
```

Comment:
By using Integer vs int you can get this behavior though in most cases not the cleanest solution